src/meshapi/static/widgets/warn_about_date.js

```diff
@@ -0,0 +1,111 @@
+$(document).ready(function($) {
+    const getCurrentISODate = () => {
+      const now = new Date();
+      const year = now.getFullYear();
+      const month = String(now.getMonth() + 1).padStart(2, '0'); // Months are 0-indexed
+      const date = String(now.getDate()).padStart(2, '0');
+      return `${year}-${month}-${date}`;
+    };
+
+    function submitAdminForm(saveAction) {
+        const adminForm = document.querySelector('#content-main form');
+
+        // Create a hidden input to simulate the button action
+        const hiddenInput = document.createElement('input');
+        hiddenInput.type = 'hidden';
+        hiddenInput.name = saveAction;
+        hiddenInput.value = '1';
+        adminForm.appendChild(hiddenInput);
+        adminForm.submit();
+
+        return true;
+    }
+
+    const saveButtons = $('.submit-row input[value*="Save"]');
+
+
+    function hideWarning(){
+        $('#dateMissingWarning').remove();
+        saveButtons.each((i, button) => {
+            $(button).removeClass("disabled-button");
+        })
+    }
+
+    function showWarning(dateField, statusField, saveAction, dateType){
+      const saveCurrentDateButton = document.createElement("a");
+        saveCurrentDateButton.className = "button";
+        saveCurrentDateButton.onclick = function () {
+            dateField.val(getCurrentISODate())
+            submitAdminForm(saveAction);
+            return false;
+        }
+        saveCurrentDateButton.style = "padding: 10px 15px; display: inline-block; text-decoration: none;";
+        saveCurrentDateButton.href = "#";
+        saveCurrentDateButton.innerText = "Use today's date";
+
+        const saveNoDateButton = document.createElement("a");
+        saveNoDateButton.className = "button";
+        saveNoDateButton.onclick = function () {
+            submitAdminForm(saveAction);
+            return false;
+        }
+        saveNoDateButton.style = "padding: 10px 15px; display: inline-block; margin-left: 10px; text-decoration: none;";
+        saveNoDateButton.href = "#";
+        saveNoDateButton.innerText = "Continue without setting date";
+
+        const errorNotice = document.createElement("div");
+        errorNotice.id = "dateMissingWarning";
+        errorNotice.className = "warning-box";
+        errorNotice.innerHTML = `
+            <p><b>Warning</b>: Status set to "${statusField.val()}" without setting ${dateType} Date</p>
+        `;
+        errorNotice.appendChild(saveCurrentDateButton);
+        errorNotice.appendChild(saveNoDateButton);
+
+        saveButtons.each((i, button) => {
+            $(button).addClass("disabled-button");
+        })
+
+        $(errorNotice).insertBefore($('.submit-row').first())
+    }
```

Comment:
Was it really easier to do this than to create another html template? I guess if it works, but this isn't terribly readable.

Reply:
I’ll try to implement a template based solution tonight and see how it looks. The nice thing about this approach is that it’s modular and could be used alongside other form widgets

Reply:
I suppose so, and I guess if it works it works...